ggml-quants.h

```diff
@@ -229,19 +229,19 @@ void dequantize_row_iq2_xxs(const block_iq2_xxs * restrict x, float * restrict y
 void dequantize_row_iq2_xs (const block_iq2_xs  * restrict x, float * restrict y, int k);
 
 // Dot product
-void ggml_vec_dot_q4_0_q8_0(int n, float * restrict s, const void * restrict vx, const void * restrict vy);
-void ggml_vec_dot_q4_1_q8_1(int n, float * restrict s, const void * restrict vx, const void * restrict vy);
-void ggml_vec_dot_q5_0_q8_0(int n, float * restrict s, const void * restrict vx, const void * restrict vy);
-void ggml_vec_dot_q5_1_q8_1(int n, float * restrict s, const void * restrict vx, const void * restrict vy);
-void ggml_vec_dot_q8_0_q8_0(int n, float * restrict s, const void * restrict vx, const void * restrict vy);
+void ggml_vec_dot_q4_0_q8_0(int n, float * restrict s, const void * vx[restrict], const void * vy[restrict], const int nrc);
+void ggml_vec_dot_q4_1_q8_1(int n, float * restrict s, const void * vx[restrict], const void * vy[restrict], const int nrc);
+void ggml_vec_dot_q5_0_q8_0(int n, float * restrict s, const void * vx[restrict], const void * vy[restrict], const int nrc);
+void ggml_vec_dot_q5_1_q8_1(int n, float * restrict s, const void * vx[restrict], const void * vy[restrict], const int nrc);
+void ggml_vec_dot_q8_0_q8_0(int n, float * restrict s, const void * vx[restrict], const void * vy[restrict], const int nrc);
```

Comment:
This code is correct according to the C standard, but it looks like MSVC will only be happy if you change all of these arrays of pointers to pointers to pointers, across all files:
```suggestion
void ggml_vec_dot_q8_0_q8_0(int n, float * restrict s, const void ** restrict vx, const void ** restrict vy, const int nrc);
```

Reply:
thanks for looking into it, @cebtenzzre . double pointer works for other compilers as well, I will update the interface.